src/logging.cc

```diff
@@ -66,82 +67,95 @@ Logger::Flush()
-              << std::setw(2) << tm_time.tm_sec << '.' << std::setw(6)
-              << tv.tv_usec << ' ' << static_cast<uint32_t>(getpid()) << ' '
-              << path << ':' << line << "] ";
-#endif
+      stream << std::setfill('0') << std::setw(2) << (tm_time.tm_mon + 1)
+             << std::setw(2) << tm_time.tm_mday << ' ' << std::setw(2)
+             << tm_time.tm_hour << ':' << std::setw(2) << tm_time.tm_min << ':'
+             << std::setw(2) << tm_time.tm_sec << '.' << std::setw(6)
+             << timestamp_.tv_usec;
       break;
     }
     case Logger::Format::kISO8601: {
-      // YYYY-MM-DDThh:mm:ssZ L
-#ifdef _WIN32
-      SYSTEMTIME system_time;
-      GetSystemTime(&system_time);
-      stream_ << system_time.wYear << '-' << std::setfill('0') << std::setw(2)
-              << system_time.wMonth << '-' << std::setw(2) << system_time.wDay
-              << 'T' << std::setw(2) << system_time.wHour << ':' << std::setw(2)
-              << system_time.wMinute << ':' << std::setw(2)
-              << system_time.wSecond << "Z "
-              << level_name_[std::min(level, (uint32_t)Level::kINFO)] << ' '
-              << static_cast<uint32_t>(GetCurrentProcessId()) << ' ' << path
-              << ':' << line << "] ";
-#else
-      struct timeval tv;
-      gettimeofday(&tv, NULL);
-      struct tm tm_time;
-      gmtime_r(((time_t*)&(tv.tv_sec)), &tm_time);
-      stream_ << (tm_time.tm_year + 1900) << '-' << std::setfill('0')
-              << std::setw(2) << (tm_time.tm_mon + 1) << '-' << std::setw(2)
-              << tm_time.tm_mday << 'T' << std::setw(2) << tm_time.tm_hour
-              << ':' << std::setw(2) << tm_time.tm_min << ':' << std::setw(2)
-              << tm_time.tm_sec << "Z "
-              << level_name_[std::min(level, (uint32_t)Level::kINFO)] << ' '
-              << static_cast<uint32_t>(getpid()) << ' ' << path << ':' << line
-              << "] ";
+      stream << (tm_time.tm_year + 1900) << '-' << std::setfill('0')
+             << std::setw(2) << (tm_time.tm_mon + 1) << '-' << std::setw(2)
+             << tm_time.tm_mday << 'T' << std::setw(2) << tm_time.tm_hour << ':'
+             << std::setw(2) << tm_time.tm_min << ':' << std::setw(2)
+             << tm_time.tm_sec << "Z";
+      break;
+    }
+  }
+}
+
 #endif
+
+void
+LogMessage::LogPreamble(std::stringstream& stream)
+{
+  switch (gLogger_.LogFormat()) {
+    case Logger::Format::kDEFAULT: {
+      stream << Logger::LEVEL_NAMES[static_cast<uint8_t>(level_)];
+      LogTimestamp(stream);
+      stream << ' ' << pid_ << ' ' << path_ << ':' << line_ << "] ";
+
+      break;
+    }
+    case Logger::Format::kISO8601: {
+      LogTimestamp(stream);
+      stream << " " << Logger::LEVEL_NAMES[static_cast<uint8_t>(level_)] << ' '
+             << pid_ << ' ' << path_ << ':' << line_ << "] ";
       break;
     }
   }
 }
 
+
 LogMessage::~LogMessage()
 {
-  gLogger_.Log(stream_.str());
+  std::stringstream preamble;
```

Comment:
Change to `output_stream` or something similar, this variable is more than a preamble